### PR TITLE
Allow User to propose alias and Admin to approve/reject

### LINKED ISSUE #23: Frontend for FEATURE #2
Create a free response text box on the "Profile" page that allows users to modify their alias #2

### PR DESCRIPTION
**Pull request by cs11111111**
Users can navigate to their profile page where their alias is displayed as Anonymous User and will find an input box to input an alias. After they input their alias, it will appear in the Users table with the status "Awaiting Moderation." Admins can approve or reject this alias through swagger. If approved, the Users table will display "Approved on [date approved]". If rejected, it will display "Rejected." Once approved, the new alias will display on the profile page. Otherwise, it will remain as Anonymous User.

Dokku deployment: https://proj-dining-cs11111111.dokku-16.cs.ucsb.edu

Closes #23

### CODE REVIEW

**Review by li-kelly — APPROVED** [review]
LGTM, my initial alias was "Anonymous User", when I inputted an Alias it pops up with a notification that it's waiting for moderation, after I approved my alisas on swagger now it appears on my profile page. My alias is also updated on the Users Table page.

File: frontend/src/fixtures/usersFixtures.js [hunk 1]
```diff
@@ -13,6 +13,9 @@ const usersFixtures = {
       locale: "en",
       hostedDomain: "ucsb.edu",
       admin: true,
+      alias: "Alias1",
+      status: "Awaiting Moderation",
+      dateApproved: null,
```
Comment (pconrad) [inline frontend/src/fixtures/usersFixtures.js:18]: Your entity has a proposedAlias field, but the fixtures do not.

Why not?

I suggest you add it.

**Review by pconrad — DISMISSED** [review]
This doesn't really think the user story all the way through.

Both the user and the admin need to be able to see the proposed alias value (beyond just the toast.)

File: frontend/src/main/pages/ProfilePage.js [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
+import { useForm } from "react-hook-form";
+import { useBackendMutation } from "main/utils/useBackend";
+import UsersTable from "main/components/Users/UsersTable";
 
-import { Inspector } from "react-inspector";
 const ProfilePage = () => {
   const { data: currentUser } = useCurrentUser();
+  const { root } = currentUser || {};
+  const { user } = root || {};
+  const { email, pictureUrl, fullName} = user || {};
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:15]: If you move these to after the check for `  if (!currentUser?.loggedIn) {` then you won't need the `|| {};` stuff which means you won't have to test it.

File: frontend/src/main/pages/ProfilePage.js [hunk 3]
```diff
@@ -1,18 +1,47 @@
...
+      proposedAlias: user.proposedAlias,
+    },
+  });
+  const onSuccess = (user) => {
+    toast(`Alias Awaiting Moderation: ${user.proposedAlias}`);
+  };
+
+  const mutation = useBackendMutation(objectToAxiosParams, { onSuccess }, ["current user"]);
+  
+  if (!currentUser?.loggedIn) {
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:35]:
```suggestion
  if (!currentUser.loggedIn) {
```

I don't think you need this if you adjust the test properly (whatever problem this sovled, there's a better solution).  And introducing it adds an extra mutation you have to cover.   So better to leave it out.

**Review by pconrad — COMMENTED** [review]
a few more suggestions based on trying a few things.;;

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**